lib/actions/os.coffee

```diff
@@ -43,9 +94,26 @@ exports.download =
 
 		console.info("Getting device operating system for #{params.type}")
 
-		manager.get(params.type).then (stream) ->
-			bar = new visuals.Progress('Downloading Device OS')
-			spinner = new visuals.Spinner('Downloading Device OS (size unknown)')
+		version = options.version
+		displayVersion = ''
+		Promise.try ->
+			if not version
+				version = 'default'
+				console.warn('OS version is not specified, using the default version:
+					the newest stable (non-pre-release) version if available,
+					or the newest version otherwise (if all existing
+					versions for the given device type are pre-release)')
+				return
+			return resolveVersion(params.type, version)
+		.then (resolvedVersion) ->
+			version = resolvedVersion
```

Comment:
There's some strange behaviour here. If `options.version` isn't set, we set `version` to 'default' above, but we then implicitly return undefined, so `resolvedVersion` is undefined, so we overwrite `version` with undefined here. That means version can never be default in the check below, and this is currently actually defaulting to `latest` behaviour (which is what `manager.get` will do with an undefined version).